src/app/qgswelcomepage.cpp

```diff
@@ -277,9 +340,86 @@ void QgsWelcomePage::showContextMenuForTemplates( QPoint point )
   menu->popup( mTemplateProjectsListView->mapToGlobal( point ) );
 }
 
+void QgsWelcomePage::showContextMenuForNews( QPoint point )
+{
+  QModelIndex index = mNewsFeedListView->indexAt( point );
+  if ( !index.isValid() )
+    return;
+
+  const int key = index.data( QgsNewsFeedModel::Key ).toInt();
+
+  QMenu *menu = new QMenu();
+
+  QAction *dismissAction = new QAction( tr( "Dismiss" ), menu );
+  connect( dismissAction, &QAction::triggered, this, [this, key]
+  {
+    mNewsFeedParser->dismissEntry( key );
+  } );
+  menu->addAction( dismissAction );
+  QAction *dismissAllAction = new QAction( tr( "Dismiss All" ), menu );
+  connect( dismissAllAction, &QAction::triggered, this, [this]
+  {
+    mNewsFeedParser->dismissAll();
+  } );
+  menu->addAction( dismissAllAction );
+  menu->addSeparator();
+  QAction *hideAction = new QAction( tr( "Hide QGIS News…" ), menu );
```

Comment:
Great integration; we could possibly add similar actions to the recent projects and template items (in a separate PR).